### PR TITLE
test/librbd: move tests using non-public api to internal

### PR DESCRIPTION
**Pull request by vshankar**
moved `TestCoR` and `FlattenNoEmptyObjects` to internal tests.

### CODE REVIEW

File: src/test/librbd/test_support.cc [hunk 1]
```diff
@@ -22,6 +22,33 @@ bool is_feature_enabled(uint64_t feature) {
   return (get_features(&features) && (features & feature) == feature);
 }
 
+int create_image_full_pp(librbd::RBD &rbd, librados::IoCtx &ioctx,
```
Comment (vshankar) [inline src/test/librbd/test_support.cc]: `create_image_pp()` below is not changed to use this helper as `DiffIterateTest/1.DiffIterate` test case fails with `stripingv2` image.

File: src/test/librbd/test_internal.cc [hunk 2]
```diff
@@ -924,3 +971,260 @@ TEST_F(TestInternal, DiffIterateCloneOverwrite) {
...
+  }
+
+  // close child to flush all copy-on-read
+  ASSERT_EQ(0, image.close());
+
+  printf("check whether child image has the same set of objects as parent\n");
+  ASSERT_EQ(0, m_rbd.open(m_ioctx, image, clonename.c_str(), NULL));
+  ASSERT_EQ(0, image.stat(info, sizeof(info)));
+
+  ASSERT_EQ(0, rados_nobjects_list_open(d_ioctx, &list_ctx));
```
Comment (vshankar) [inline src/test/librbd/test_internal.cc]: Ideally we'd use `nobejcts_begin()`, `nobjects_end()` and `get_next()` methods from `librados::IoCtx`, but it seems like these methods are not implemented in librados test stub.

**Comment by dillaman** [comment]
retest this please

**Comment by dillaman** [comment]
Passed teuthology tests: http://pulpito.ceph.com/jdillaman-2017-03-06_18:43:47-rbd-wip-jd-testing-distro-basic-smithi/

**Pull request merged**